src/marqo/core/index_management/index_management.py

```diff
@@ -158,51 +182,62 @@ def batch_create_indexes(self, marqo_index_requests: List[MarqoIndexRequest]) ->
+                if not configured:
+                    for index in marqo_index_requests:
+                        if self.index_exists(index.name):
+                            raise IndexExistsError(f"Index {index.name} already exists")
 
-        for schema, marqo_index in schema_responses:
-            logger.debug(f'Creating index {str(marqo_index)} with schema:\n{schema}')
-            self._add_schema(app, marqo_index.schema_name, schema)
-            self._add_schema_to_services(app, marqo_index.schema_name)
+                # Populate the prefix fields if they are None
+                for index in marqo_index_requests:
+                    if index.model.text_query_prefix is None:
+                        index.model.text_query_prefix = index.model.get_default_text_query_prefix()
+                    if index.model.text_chunk_prefix is None:
+                        index.model.text_chunk_prefix = index.model.get_default_text_chunk_prefix()
 
-        self.vespa_client.deploy_application(app)
+                schema_responses = [
+                    vespa_schema_factory(index).generate_schema()  # Tuple (schema, MarqoIndex)
+                    for index in marqo_index_requests
+                ]
 
-        self.vespa_client.wait_for_application_convergence()
+                for schema, marqo_index in schema_responses:
+                    logger.debug(f'Creating index {str(marqo_index)} with schema:\n{schema}')
+                    self._add_schema(app, marqo_index.schema_name, schema)
+                    self._add_schema_to_services(app, marqo_index.schema_name)
+
+                self.vespa_client.deploy_application(app)
 
-        for _, marqo_index in schema_responses:
-            self._save_index_settings(marqo_index)
+                self.vespa_client.wait_for_application_convergence()
 
-        return [schema_resp[1] for schema_resp in schema_responses]
+                for _, marqo_index in schema_responses:
+                    self._save_index_settings(marqo_index)
+
+                return [schema_resp[1] for schema_resp in schema_responses]
+        except ZooKeeperLockNotAcquiredError:
+            raise IndexCreationAndDeletionConflictError("Another index creation/deletion operation is in progress. "
+                                                        "Your batch_create_indexes request is rejected. "
+                                                        "Please try again later.")
 
     def delete_index(self, marqo_index: MarqoIndex) -> None:
         """
-        Delete a Marqo index.
+        Delete a Marqo index. To make this operation thread-safe, use delete_index_by_name instead.
 
         This method is idempotent and does not raise an error if the index does not exist.
 
         Args:
             marqo_index: Marqo index to delete
         """
-        app = self.vespa_client.download_application()
+        app = self.vespa_client.download_application(wait_for_application_convergence=True)
```

Comment:
This should be `check_for_application_convergence` and error out with concurrency exception if hasn't converged. I wouldn't wait for convergence here

Reply:
Done. I add a 1-second wait though